Assets/Scripts/Explodable/ExplodableObject.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ExplodableObject : MonoBehaviour
+{
+
+    private float m_TerrainRevealRadius = 3.0f;
+
+    public const float STANDARD_BOMB_DELAY  = 3.0f; // Constnat Value
+
+    private float m_ExecutionDelay;
+
+    private bool m_hasExploded = false;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        m_ExecutionDelay = STANDARD_BOMB_DELAY;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        m_ExecutionDelay -= Time.deltaTime;
+        if (m_ExecutionDelay <= 0 && !m_hasExploded) {
+            m_hasExploded = true;
+            Explode();
+            Destroy(gameObject);
+        }
+    }
```

Comment:
Instead of doing physics calls every frame, could we let the physics engine do the check instead? I.e., Use the methods `OnTriggerEnter` or `OnCollisionEnter`. We do checks every frame in the `Reveal*` scripts because we have a huge number of objects such as floors where large triggers would not be clean. In the case of bombs or small explodable objects, let's use `OnTriggerEnter` and check for the player in there instead.

Reply:
That's a great idea!

So maybe when a player hits the collider / collision box, we can have the bomb countdown then 💥